### ISSUE TITLE
Using .freeze() in nlp.plugin()? 

### ISSUE THREAD

**Issue by thegoatherder** [spencermountain/compromise#1080]
The documentation for [`.freeze()`](https://observablehq.com/@spencermountain/compromise-freeze) shows adding a word to the lexicon and freezing it using `addWords()`.

For projects that use the [`nlp.plugin()`](https://observablehq.com/@spencermountain/compromise-plugins) option to initialise the lexicon at program start, an object is passed to the `words` property with a name-value pair syntax for each tag.

Is there a way to specify these tags to be frozen from within the plugin object?

**Comment by thegoatherder** [comment]
Can it also be specified in a match-object?

```javascript
{
  match: "#Diagnostic+ (preassessment|assessment|reassessment|pre-assessment|re-assessment)",
  reason: "",
  tag: "Diagnostic",
}
```

**Comment by spencermountain** [comment]
Hey Adam, good idea - I’ll look at adding both of these features over the
next few days.
Let me know if there’s anything else you can find
Cheers

**Comment by thegoatherder** [comment]
Thanks, Spence! Once these are implemented we can fully integrate it into our solution and get it tested thoroughly. We will let you know if we find any anomalies! Looking forward to getting this feature rolling…!

**Comment by spencermountain** [comment]
hey Adam, both features are implemented on dev, and i will test, document, and release it, hopefully tomorrow.
cheers

**Comment by thegoatherder** [comment]
Hey Spence that’s great! I forgot to mention the reason we need the match-object is for buildNet() and sweep() - will these support frozen tags too?

**Comment by spencermountain** [comment]
yep - `{match:'foo', tag:'Bar', freeze:true, }` will lock in #Bar, and it will be unchangeable. Will that do the trick for your purpose?

**Comment by thegoatherder** [comment]
Yep absolutely perfect, thank you!

**Comment by spencermountain** [comment]
both features have been released in `14.2.0`:

frozen lexicon via plugin:
```js
nlp.plugin({
  // normal lexicon
  words:{
    foo:'Bar'
  },
  // frozen lexicon
  frozen: {
    'juicy fruit': 'Singular',
    'front steps': 'Plural',
  },
})
let doc = nlp(`i ate juicy fruit on the front steps`)
doc.debug()
```

and `freeze` inside sweep:
```js
let matches = [
  { match: 'juicy fruit', tag: 'Singular', freeze: true },
  { match: 'front steps', tag: 'Plural', freeze: true },
]
let doc = nlp(`i ate juicy fruit on the front steps`)
let net = nlp.buildNet(matches)
doc.sweep(net)
doc.debug()
```

note that in both cases, the words don't stay frozen, after this process. You can do `doc.sweep(net).freeze()` to re-freeze them, for further analysis.
cheers

**Comment by thegoatherder** [comment]
Superb thanks Spence.
@Fdawgs is going to integrate this on our end in the coming days and we will report back on any anomalies found during testing. Thanks again

**Comment by Fdawgs** [comment]
> both features have been released in `14.2.0`:

Did you mean `14.11.2` @spencermountain?
